tests/test_xarray.py

```diff
@@ -114,8 +114,9 @@ def test_convert_units(test_var):
     assert units(test_var.attrs['units']) == units('degC')
```

Comment:
@jthielen Can you sanity check me on this `test_convert_units` test? I can't see any way that this was correct, at least in the days before we had the integration between xarray and pint working like it does now. It was confusing to see the test suddenly fail, and I want to make sure that I'm right that it should have been failing all along.

Reply:
The diff in this test here doesn't seem to be doing anything meaningful, unless something is broken under the hood in `assert_almost_equal`. Does this diff make a difference between passing and failing?

Since I'm not completely sure what is the direct cause for confusion, here are some notes below:

This particular line

```python
assert units(test_var.attrs['units']) == units('degC')
```

was updated back in https://github.com/Unidata/MetPy/pull/1144 since Pint changed its canonical string for °C from `degC` to `degrees_Celsius`, so we can't do string comparison, but comparison of units is still fine.

But, this whole test is no longer valid after Quantity-in-xarray was introduced and after we moved away from in-place unit conversions, hence why this test on master is currently:

```python
def test_convert_units(test_var):
    """Test conversion of units."""
    result = test_var.metpy.convert_units('degC')

    # Check that units are updated without modifying original
    assert result.metpy.units == units.degC
    assert test_var.metpy.units == units.kelvin

    # Make sure we now get an array back with properly converted values
    assert_almost_equal(result[0, 0, 0, 0], 18.44 * units.degC, 2)
```

This test definitely relies on `assert_almost_equal` automatically converting DataArrays to their corresponding unit arrays, hence why it should have worked fine before this diff, and I don't see what the diff in effect changed.

Reply:
Well, it *should* rely on that. That test *did* fail on one of the earlier runs. The test passes on 0.12.1. It turns out, though, that the only reason that test passed was because we were swallowing an `AttributeError` when calling `.to()` on a `DataArray`. I think the right way to do this for a bugfix release is to put back the `AttributeError` clause in `check_and_drop_units()` since that change wasn't necessary (and I'm not sure where it snuck in).

Reply:
Ah, the change to testing crept in on 771cbf0. I'll pop that off and my most recent change to the test.